cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -563,365 +542,18 @@ impl<T: Config> Pallet<T> {
 			s.push(OutboundChannelDetails::new(dest).with_signals());
 		}
 		<SignalMessages<T>>::mutate(dest, |page| {
-			if page.is_empty() {
-				*page = (XcmpMessageFormat::Signals, signal).encode();
-			} else {
-				signal.using_encoded(|s| page.extend_from_slice(s));
-			}
+			*page = (XcmpMessageFormat::Signals, signal).encode();
```

Comment:
The code is just mostly legacy. I only do some small cleanup when its possible, but dont want to inflate the merge request any more.  
Yes it could be optimized a lot, also the case that one signal is send per page is very wasteful.